### DISCUSSION TITLE
[@types/chrome] - chrome.runtime.onInstall callback parameter details.reason is typed as string (should be OnInstalledReason "enum")

### DISCUSSION THREAD

**Discussion by lbragile** [DefinitelyTyped/DefinitelyTyped#57513] [Issues with a @types package] [answered]
When checking the online [documentation](https://developer.chrome.com/docs/extensions/reference/runtime/#event-onInstalled) for `chrome.runtime.onInstalled`, I see that the callback argument is typed as:

```typescript
interface object {
  id?: string;
  previousVersion?: string;
  reason: OnInstalledReason;
}

(details: object) => void
```

Where `onInstalledReason` should be an enum type (based on the [documentation](https://developer.chrome.com/docs/extensions/reference/runtime/#type-OnInstalledReason)):

```typescript
type onInstallReason = "install" | "update" | "chrome_update" | "shared_module_update";
```

However, when looking at the [actual type definitions](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/0a9c9d85e8a9ae8aa4a96abbb2d9feac53dda9f6/types/chrome/index.d.ts#L6506) found in `node_modules`, I see (comments excluded):
```typescript
// node_modules/@types/chrome/index.d.ts

export interface InstalledDetails {
    reason: string;
    previousVersion?: string | undefined;
    id?: string | undefined;
}
```

In the above, `InstalledDetails.reason` is typed as a `string`, which does not provide intellisense like an enum type would.
Furthermore, I cannot seem to find the `OnInstalledReason` enum type in the file.

I am using @types/chrome@0.0.159 but the same is present in v0.0.164 (latest)

Would a PR for this be helpful?

[a comment by an automated account is not shown]

**Comment by nicolas377** [comment, marked as answer]
Hey @lbragile! I have a PR for this, #57504, up already. Unfortunately, I seem to be one of the only active maintainers, so it might get held up for a while.

**Reply by lbragile** [reply]
Hi @nicolas377,

Awesome, that is nice to see.

Understandable, thank you for your work!
